model/SearchEngine/Service/ItemUsageService.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2025 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoAdvancedSearch\model\SearchEngine\Service;
+
+use oat\tao\model\search\ResultSet;
+use oat\taoAdvancedSearch\model\SearchEngine\Driver\Elasticsearch\ElasticSearch;
+
+class ItemUsageService
```

Comment:
We do not need this service, we just need to call `search` method https://github.com/oat-sa/extension-tao-advanced-search/pull/140/files#diff-3af9fe84e20793c98cd35995836e8b435d33b81237207734ce7cdd019a62f363L95 on the new `ItemRelationsService`. No need to complicate things

Reply:
Hi @bartlomiejmarszal I really think we do not need this service and we can do this query directly inside `ItemRelationsService`.